mono/mini/mini-runtime.c

```diff
@@ -1905,6 +1905,97 @@ mono_jit_map_is_enabled (void)
 
 #endif
 
+#if ENABLE_JIT_DUMP
+static FILE* perf_dump_file;
+uint64_t codeIndex = 0;
+pthread_mutex_t mutex = PTHREAD_MUTEX_INITIALIZER;
+void *mmapAddr = MAP_FAILED;
+
+void
+mono_enable_jit_dump (void)
+{
+        if (!perf_dump_file) {
+		char name [64];
+		FileHeader * header = malloc (sizeof (FileHeader));
+		int result = 0;
+		
+		result = pthread_mutex_lock (&mutex);
+		
+		g_snprintf (name, sizeof (name), "/tmp/jit-%d.dump", getpid ());
+		unlink (name);
+		perf_dump_file = fopen (name, "w");
```

Comment:
You might want to check these for possible errors -- for instance, if the file could not be unlinked (say, because another user owned it), and if `fopen()` fails to open the file (for the same reason), calls to `fwrite()` will be made with a NULL pointer.

Reply:
Also, since stdio is being used to write to these files, there's a chance that the buffer is not flushed to the disk when the process ends. This might or might not be an issue; I'm not familiar enough with the mini-runtime. Might be a good idea to call `fflush()` and `flose()` in this file before `main()` finishes to avoid having truncated dump files if stdio had buffers large enough that they never got flushed to disk.

An option is to use file descriptors directly, and then using `writev()` to write each record; has more syscall overhead, but there's no overbuffering in mini-runtime.

Reply:
See also `setvbuf`.

Reply:
Looks like still missing error check on the `fopen`.